Assets/Scripts/Player/Body/Body.cs

```diff
@@ -101,5 +110,30 @@ private BodyPart FindBodyPart(BodyPartType bodyPartType)
 		{
 			return bodyParts.FirstOrDefault(bodyParts => bodyParts.BodyPartType == bodyPartType);
 		}
-	}
+
+        private void UpdateBodyPartList()
+        {
+            bodyParts.Clear();
+            foreach (BodyPart part in gameObject.GetComponentsInChildren<BodyPart>())
+            {
+                bodyParts.Add(part);
+            }  
+        }
+
+        private void UpdateBodyPartVisuals(GameObject newBodyPart, BodyPart bodyPart)
+        {
+            // Making sure the part is equal to the original one
+            SkinnedMeshRenderer newBodyPartMesh = newBodyPart.GetComponent<SkinnedMeshRenderer>();
```

Comment:
Just parsing in `SkinnedMeshRenderer newBodyPartMeshRenderer` is cleaner than parsing in a GameObject and getting the component. Or maybe you actually want to have it `BodyPart` like the other parameter is and the name suggests.
With you current style you have no safety and you aren't checking if the component exist.